client/lib/performance-tracking/lib.js

```diff
@@ -40,23 +40,28 @@ const getDefaultCollector = ( state ) => {
 	};
 };
 
+const buildMetadataCollector = ( metadata = {} ) => {
+	return ( reporter ) => {
+		Object.entries( metadata ).forEach( ( [ key, value ] ) => reporter.data.set( key, value ) );
+	};
+};
+
 const isPerformanceTrackingEnabled = () => {
 	const isEnabledForEnvironment = config.isEnabled( CONFIG_NAME );
 	const isEnabledForCurrentInteraction = abtest( AB_NAME ) === AB_VARIATION_ON;
 	return isEnabledForEnvironment && isEnabledForCurrentInteraction;
 };
 
-export const startPerformanceTracking = (
-	name,
-	{ fullPageLoad = false, collectors = [] } = {}
-) => {
+export const startPerformanceTracking = ( name, { fullPageLoad = false } = {} ) => {
 	if ( isPerformanceTrackingEnabled() ) {
-		start( name, { fullPageLoad, collectors } );
+		start( name, { fullPageLoad } );
 	}
 };
 
-export const stopPerformanceTracking = ( name, state, { collectors = [] } = {} ) => {
```

Comment:
No more passing `collectors`, so we don't leak the reporter API outside this wrapper. I imagine most future consumers will be ok with just passing a set of key/value pairs to include in the report.